[RFT,RFC] generic: Add support for 6.1 kernel version

Ansuel opened the pull request:
Hi, just to be sure about this, some prereq:
- ~~99% kernel 6.1 will be the next LTS version~~ CONFIRMED
- ~~openwrt still needs to stabilize and switch everything to 5.15 and AFAIK we don't really support having 3 different kernel version in our master repo.~~ We can finally merge this
- This require more work and it's really experimental...
- ~~We need to wait for 6.1 to be actually released... this is based on rc1.~~

Generic todo
---

With these prereq here the TODO:
- Correctly refresh config
- Fix any regression with kernel modules 
- Test with other target
- Refresh all of this when 6.1 will actually be released

Todo unrelated
---
These TODO are good to know but should be handled later with package update or on target specific bump. (lets not make this pr too big) (can be handled anyway if i'm crazy enough to have time for them)
- `error: too many arguments to function 'netif_napi_add'`
  6.1 https://github.com/torvalds/linux/commit/b48b89f9c189d24eb5e2b4a0ac067da5a24ee86d net: drop the weight argument from netif_napi_add 
  - package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_adsl.c
  - target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c
  - mac80211 backports
- `error: implicit declaration of function 'netif_tx_napi_add'; did you mean 'netif_napi_add'`
  6.1 https://github.com/torvalds/linux/commit/c3f760ef128789252e7c4f10d3c1721422dceba9 net: remove netif_tx_napi_add()
  from 5.19 https://github.com/torvalds/linux/commit/58caed3dacb4354a25a1aa8d2febc3e9648ba1f4 netdev: reshuffle netif_napi_add() APIs to allow dropping weight
  - target/linux/ipq40xx/files/drivers/net/ethernet/qualcomm/ipqess/ipqess.c
  - mac80211 backports

---

feeds packages with kmods

- coova-chilli https://github.com/openwrt/packages/pull/19654
  5.17 https://github.com/torvalds/linux/commit/359745d78351c6f5442435f81549f0207ece28aa
- libpfring   https://github.com/openwrt/packages/pull/19653
  5.16 https://github.com/torvalds/linux/commit/887069f424550ebdcb411166733e1d05002b58e4, 5.17 https://github.com/torvalds/linux/commit/359745d78351c6f5442435f81549f0207ece28aa, 5.18 https://github.com/torvalds/linux/commit/2655926aea9beea62c9ba80c032485456fd848f0

---

This was really helpful as there are many patch that should be moved from pending to backports as they got merged upstream.

---

Changelog
16-09-2022: First rebase targeting 6.0-rc5
18-10-2022: First rebase targeting 6.1-rc1
20-10-2022: Refresh config, refresh kernel modules, rebase on top of master, address some comments
21-10-2022: Refresh kernel modules, merge robi suggestion patch
22-10-2022: Other fixup, created a pr to move 5.15 pending patch to backport to indirectly reduce delta of this pr, restore more patch that was removed by mistake
26-10-2022: Refresh on top of master, updated to 6.1-rc2, fix patch. drop backport commit
13-12-2022: Refresh on top of master, update to 6.1, fix patch, drop backport commit, add missing config
22-05-2023: Refresh on top of master, update to 6.1.29, fix patch, drop backport commit, general refresh in preparation for merge.

Ansuel committed:
kernel: add define for kernel 6.1

Add define for kernel 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: copy backport, hack, pending patch and config from 5.15 to 6.1

Copy backport, hack, pending patch and config from 5.15 to 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: drop backport patches

Drop all backport patches that are now included in kernel 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: manually refresh pending patches

Refresh pending patches for kernel 6.1.

Changes:
- Refresh mtd patches with new implementation.
- Change 191-rtc-rs5c372-let_the_alarm_to_be_used_as_wakeup_source as
  uie_unsupported got dropped and we now set the bit directly.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: manually refresh hack patches

Refresh hack patches for kernel 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: refresh config and add missing symbols

Refresh generic config for 6.1 and add missing symbols.
Then sorted using:
$ ./scripts/kconfig.pl '+' target/linux/generic/config-6.1 /dev/null > target/linux/generic/config-6.1-new
$ mv target/linux/generic/config-6.1-new target/linux/generic/config-6.1

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: 6.1: copy patches, files and config from 5.15

Copy patches, files and config from 5.15 kernel version.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: 6.1: drop upstream patch

Drop patch that got merged upstream.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: 6.1: replace ZRELADDR HACK patch with new version

With 6.1 lots have changed and the platform for Makefile.boot got
dropped. Replace the patch with a new version that is alligned to the
new implementation.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: 6.1: refresh config and add missing symbols

Refresh config for kernel 6.1 and add missing symbols
Then sorted using:
$ ./scripts/kconfig.pl '+' target/linux/ipq806x/config-6.1 /dev/null > target/linux/ipq806x/config-6.1-new
$ mv target/linux/ipq806x/config-6.1-new target/linux/ipq806x/config-6.1

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
kernel: add CONFIG_INITRAMFS_PRESERVE_MTIME config to Initramfs function

New kernel version 6.1 introduced new INITRAMFS option. Add them to the
Initramfs functions to correctly compile initramfs images.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: enable testing kernel version with 6.1 version

Enable testing kernel version with 6.1 kernel version.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: manually refresh backport patches

Refresh backport patches that still needs to be merged for kernel 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: 6.1: manually refresh kernel patches

Refresh kernel patches for version 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: add pending OPP set opp improvement patch

Add pending OPP set opp improvement patch to better handle _set_opp
function in clock scaling.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

john-tho commented:
Huge thanks for starting this Ansuel.

Can we look into the two patches left in backports. If we need them, and whether they should be somewhere else?
- [011-kbuild-export-SUBARCH.patch](https://github.com/openwrt/openwrt/blob/master/target/linux/generic/backport-5.15/011-kbuild-export-SUBARCH.patch)
  pending? In the mail at least reasoning is detailed, so we can check if still needed: https://lore.kernel.org/all/20170206093142.64537-1-nbd@nbd.name/

- [343-netfilter-nft_flow_offload-handle-netdevice-events-f.patch](https://github.com/openwrt/openwrt/blob/master/target/linux/generic/backport-5.15/343-netfilter-nft_flow_offload-handle-netdevice-events-f.patch)
  maybe hack? Introduced in https://github.com/openwrt/openwrt/commit/103335644265d96c656a7de3d5994fbd11246300#diff-f3e0a9cb582eaaf7967e351cdd010e93b71964b74a7bcc12075cb8fdb8856aa9
  I cannot find mention of it otherwise: no [`nf_flow_table_netdev_event` in lore.kernel.org](https://lore.kernel.org/all/?q=nf_flow_table_netdev_event), and few results [from:pablo flow_offload_netdev_event](https://lore.kernel.org/all/?q=f%3Apablo%40netfilter.org+flow_offload_netdev_event)

Ansuel commented:
@john-tho as you can see i also made specific commit for patch in pending that got dropped as they were present upstream... so it would be a good idea to cleanup current 5.10 and 5.15 with the new info we have from 6.1

yes i would move one to pending and the other to hack...

Ansuel committed:
kernel: refresh IFF_NO_IP_ALIGN ifdef for kernel >= 6.1

Kernel 6.1 expanded the priv_flags to long bitmap so update and restore
it to priv_flags.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@john-tho thanks a lot for your review... i did some of you changes

Ansuel committed:
kernel: refresh swconfig_leds to handle new trigger locking

Refresh swconfig_leds source to handle new trigger locking that is now a
spinlock.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
kernel: modules: adapt for kernel 6.1

Adapt kernel modules to support kernel 6.1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: add patch to compile ieee8211 headers

With kernel 6.1 ieee2111_ptr is not compiled by default. Add pending
patch to restore this to make backports project compatible.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

john-tho commented on package/kernel/linux/modules/netfilter.mk:
```diff
@@ -1148,21 +1148,44 @@ endef
+define KernelPackage/nft-offload-ipv4
+  SUBMENU:=$(NF_MENU)
+  TITLE:=Netfilter nf_tables routing/NAT offload support (ipv4)
+  DEPENDS:=!@LINUX_6_1 +kmod-nf-flow +kmod-nft-nat
+  KCONFIG:= CONFIG_NF_FLOW_TABLE_IPV6@lt5.17
+  FILES:= $(LINUX_DIR)/net/ipv4/netfilter/nf_flow_table_ipv4.ko@lt5.17
+  HIDDEN:=y
+  AUTOLOAD:=$(call AutoProbe,nf_flow_table_ipv4)
+endef
+
+$(eval $(call KernelPackage,nft-offload-ipv4))
+
+
+define KernelPackage/nft-offload-ipv6
+  SUBMENU:=$(NF_MENU)
+  TITLE:=Netfilter nf_tables routing/NAT offload support (ipv6)
+  DEPENDS:=!@LINUX_6_1 @IPV6 +kmod-nf-flow +kmod-nft-nat
```
I think the dependency invert needs to be the other way around: `@!`.

Also, `-ipv4` Kconfigs IPV6

john-tho commented on package/kernel/linux/modules/netsupport.mk:
```diff
@@ -91,7 +91,9 @@ define KernelPackage/vxlan
   KCONFIG:=CONFIG_VXLAN
-  FILES:=$(LINUX_DIR)/drivers/net/vxlan.ko
+  FILES:= \
+	$(LINUX_DIR)/drivers/net/vxlan.ko@le5.15 \
+	$(LINUX_DIR)/drivers/net/vxlan/vxlan.ko@ge6.1
```
vxlan move happend 5.18: https://github.com/torvalds/linux/commit/6765393614ea8e2c0a7b953063513823f87c9115

john-tho commented on package/kernel/linux/modules/other.mk:
```diff
@@ -56,7 +56,8 @@ define KernelPackage/bluetooth
 	$(LINUX_DIR)/drivers/bluetooth/btusb.ko \
-	$(LINUX_DIR)/drivers/bluetooth/btintel.ko
+	$(LINUX_DIR)/drivers/bluetooth/btintel.ko \
+	$(LINUX_DIR)/drivers/bluetooth/btmtk.ko@ge6.1
```
btmtk added in 5.17: https://github.com/torvalds/linux/commit/8c0d17b6b06c5bef45de7e08c5c3cab8367f6cbc

robimarko commented on target/linux/ipq806x/patches-6.1/0060-ARM-mach-qcom-fix-support-for-ipq806x.patch:
```diff
@@ -0,0 +1,67 @@
+ 	bool "Enable support for MDM9615"
+ 	select CLKSRC_QCOM
+ 
++config ARCH_IPQ806X
```
@Ansuel Will this break all of the checks for `ARCH_QCOM` that some drivers have or is `ARCH_QCOM` still y

robimarko replied:
Its clearly an improvement, I am just interested if it keeps `ARCH_QCOM=y` as there is a bunch of drivers that depend on it.

robimarko replied:
Perfect then

robimarko commented on target/linux/generic/pending-6.1/731-net-permit-ieee80211_ptr-even-with-no-CFG82111-suppo.patch:
```diff
@@ -0,0 +1,68 @@
+From 686c603f67ae87bf21a61b5e4b1564443f41c3ee Mon Sep 17 00:00:00 2001
+From: Christian Marangi <ansuelsmth@gmail.com>
+Date: Thu, 20 Oct 2022 03:34:43 +0200
+Subject: [PATCH] net: permit ieee80211_ptr even with no CFG82111 support
```
Hm, this should probably just be added to the backports so that `include/linux/netdevice.h` is backported as well

robimarko replied:
Ughh, not really as `ieee80211_ptr` is used outside of backports as well

Ansuel replied:
it was a nice discovery... no idea if this will be accepted upstream...

robimarko replied:
Well, from the upstream perspective I dont see why would they accept it as its guarded with a kernel symbol already.

Ansuel replied:
@hauke what to see a fun thing? check this patch....

Ansuel committed:
ipq806x: 6.1: refresh dts

Refresh dts for kernel 6.1 support.
Changes:
- nbg6817 drop amba node and reference directly sdcc1.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

robimarko committed:
kernel: modules: wwan: adapt for kernel 6.1

Update the WWAN kmods for compilation under kernel 6.1:
* Depend on kernel not being 5.10 rather than only 5.15
* Enable CONFIG_WWAN_DEBUGFS as its now optional from 5.17
* Add missing symbols for new WWAN drivers to generic config

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
@Ansuel Here are the fixes for MHI and WWAN modules under 6.1:
https://github.com/robimarko/openwrt/commit/b5d27c65071e120b0a160155f137c580b61cae9c
and
https://github.com/robimarko/openwrt/commit/e7d45a06a034f881d5ea3ac5b9fdc0168df98f3d

Ansuel commented:
Oh god what is the amount of kmods that depends on kernel 5_15....

john-tho commented:
More kmods:

- 6.1 https://github.com/torvalds/linux/commit/637a642f5ca5e850186bb64ac75ebb0f124b458d
  lib.mk Package kmod-lib-zstd is missing dependencies for the following libraries: zstd_common.ko
- 5.18 https://github.com/torvalds/linux/commit/f53207017fd99aaeecef6e61b4a3d8e868f53756
  hwmon.mk Package kmod-hwmon-adt7410 is missing dependencies for the following libraries: regmap-core.ko
- 6.1 https://github.com/torvalds/linux/commit/e0bffe3e689415ddd926c7f207e4186c8d355ed8
  usb.mk Package kmod-usb-net-asix is missing dependencies for the following libraries: phylink.ko

---

git grep for kernel changes I ran into:

- `error: too many arguments to function 'netif_napi_add'`
  6.1 https://github.com/torvalds/linux/commit/b48b89f9c189d24eb5e2b4a0ac067da5a24ee86d net: drop the weight argument from netif_napi_add 
  - package/kernel/lantiq/ltq-ptm/src/ifxmips_ptm_adsl.c
  - target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c
  - mac80211 backports
- `error: implicit declaration of function 'netif_tx_napi_add'; did you mean 'netif_napi_add'`
  6.1 https://github.com/torvalds/linux/commit/c3f760ef128789252e7c4f10d3c1721422dceba9 net: remove netif_tx_napi_add()
  from 5.19 https://github.com/torvalds/linux/commit/58caed3dacb4354a25a1aa8d2febc3e9648ba1f4 netdev: reshuffle netif_napi_add() APIs to allow dropping weight
  - target/linux/ipq40xx/files/drivers/net/ethernet/qualcomm/ipqess/ipqess.c
  - mac80211 backports

mac80211 backports kernel changes: What is the best approach here? Build a new backports package from 6.1rc1?

- `error: assignment of read-only location '*ndev->dev_addr'`
  ? `constant netdev->dev_addr`
  https://github.com/torvalds/linux/commit/406f42fa0d3cbcea3766c3111d79ac5afe711c5b
  `backports-5.15.58-1/drivers/net/wireless/ath/ath6kl/cfg80211.c`
  fix: https://github.com/torvalds/linux/commit/f2e2a083be8a99e7141d7f2ccaa78fba771816e1. For clean apply; the ath6kl part from this: https://github.com/torvalds/linux/commit/708884e7f7f3adb00ddb32d1c1a772d30bf86747
- 5.18 https://github.com/torvalds/linux/commit/2655926aea9beea62c9ba80c032485456fd848f0 net: Remove netif_rx_any_context() and netif_rx_ni()
- backports-5.15.58-1/drivers/net/wireless/ath/ath6kl/txrx.c
  https://github.com/torvalds/linux/commit/1cd2ef9fcb67115030b20ea2b560635efce746da

---

feeds packages with kmods

- coova-chilli https://github.com/openwrt/packages/pull/19654
  5.17 https://github.com/torvalds/linux/commit/359745d78351c6f5442435f81549f0207ece28aa
- libpfring   https://github.com/openwrt/packages/pull/19653
  5.16 https://github.com/torvalds/linux/commit/887069f424550ebdcb411166733e1d05002b58e4, 5.17 https://github.com/torvalds/linux/commit/359745d78351c6f5442435f81549f0207ece28aa, 5.18 https://github.com/torvalds/linux/commit/2655926aea9beea62c9ba80c032485456fd848f0

---

drop-from-pending-merged-upstream commits:
Checked the commit title to find an upstream commit ID; some may be wrong. Two I couldn't find upstream, and spotted some fixes here and there.

- missing? 484-mtd-spi-nor-add-esmt-f25l16pa.patch
  https://lore.kernel.org/all/?q=0x8c2115
- missing? 487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch
  https://lore.kernel.org/all/20210908201624.237634-1-bert@biot.com/
- missing/alternate implementation? 768-net-dsa-mv88e6xxx-Request-assisted-learning-on-CPU-port.patch
  https://lore.kernel.org/netdev/20210116012515.3152-8-tobias@waldekranz.com/   https://lore.kernel.org/netdev/20210224114350.2791260-14-olteanv@gmail.com/

- v6.1 020 multigenerational lru
  first series commit:  https://github.com/torvalds/linux/commit/e1fd09e3d1dd4a1a8b3b33bc1fd647eee9f4e475 last series commit: https://github.com/torvalds/linux/commit/8be976a0937a18118424dd2505925081d9192fd5
  fixes: https://github.com/torvalds/linux/commit/14aa8b2d5c2ebead01b542f62d68029023054774
- v5.16 309-MIPS-Add-CPU-option-reporting-to-proc-cpuinfo
  https://github.com/torvalds/linux/commit/626bfa03729959ea9917181fb3d8ffaa1594d02a
  fixes: https://github.com/torvalds/linux/commit/1cab5bd69eb1f995ced2d7576cb15f8a8941fd85

- v5.16 050-dtc-checks-Drop-interrupt-provider-address-cells-check
  https://github.com/torvalds/linux/commit/a77725a9a3c5924e2fd4cd5b3557dd92a8e46f87  DTC https://git.kernel.org/pub/scm/utils/dtc/dtc.git/commit/?id=d8d1a9a77863a8c7031ae82a1d461aa78eb72a7b
- v5.19 483-mtd-spinand-add-support-for-xtx-xt26g0xa.patch
  https://github.com/torvalds/linux/commit/f4c5c7f9d2e5ab005d57826b740b694b042a737c
- v6.1 731-net-dsa-mt7530-add-support-for-in-band-link-status.patch
  https://github.com/torvalds/linux/commit/e19de30d20809af3221ef8a2648b8a8a52e02d90
- v6.1 SFP HALNy
  - 774--net-sfp-redo-soft-state-polling.patch https://github.com/torvalds/linux/commit/8475c4b70b040f9d8cbc308100f2c4d865f810b3
  - 770-net-sfp-move-quirk-handling-into-sfp.c.patch https://github.com/torvalds/linux/commit/23571c7b96437483d28a990c906cc81f5f66374e
  - 771-net-sfp-move-Alcatel-Lucent-3FE46541AA-fixup.patch https://github.com/torvalds/linux/commit/275416754e9a262c97a1ad6f806a4bc6e0464aa2
  - 772-net-sfp-move-Huawei-MA5671A-fixup.patch https://github.com/torvalds/linux/commit/5029be761161374a3624aa7b4670174c35449bf5
  - 773-net-sfp-add-support-for-HALNy-GPON-SFP.patch & 775-net-sfp-change-HALNy-to-ignore-hardware-pins.patch https://github.com/torvalds/linux/commit/73472c830eae5fce2107f7f086f1e6827d215caf
- v6.0 781-net-ethernet-mtk_eth_soc-fix-mask-of-RX_DMA_GET_SPOR.patch https://github.com/torvalds/linux/commit/c9da02bfb1112461e048d3b736afb1873f6f4ccf
- v6.1 782-net-ethernet-mtk_eth_soc-fix-state-in-__mtk_foe_entr.patch https://github.com/torvalds/linux/commit/ae3ed15da5889263de372ff9df2e83e16acca4cb
- v6.1 825-Bluetooth-btusb-Add-a-new-VID-PID-0e8d-0608-for-MT79.patch https://github.com/torvalds/linux/commit/be55622ce673f9692cc15d26d77a050cda42a3d3
- v5.17 842-net-qmi_wwan-add-ZTE-MF286D-modem-19d2-1485.patch https://github.com/torvalds/linux/commit/078c6a1cbd4cd7496048786beec2e312577bebbf
- v5.18 aardvark PCI
  - 851-0001-phy-marvell-phy-mvebu-a3700-comphy-Remove-port-from-.patch https://github.com/torvalds/linux/commit/4bf18d5a2dd02db8c5b16a2cfae513510506df5b
  - 851-0002-phy-marvell-phy-mvebu-a3700-comphy-Add-native-kernel.patch https://github.com/torvalds/linux/commit/934337080c6c59b75db76b180b509f218640ad48
    fixes: https://github.com/torvalds/linux/commit/0a6fc70d76bddf98278af2ac000379c82aec8f11
  - 851-0003-arm64-dts-marvell-armada-37xx-Add-xtal-clock-to-comp.patch https://github.com/torvalds/linux/commit/73a78b6130d9e13daca22b86ad52f063b9403e03
  - 851-004 https://github.com/torvalds/linux/commit/ee995101fde67f85a3cd4c74f4f92fc4592e726b
  - 851-005 https://github.com/torvalds/linux/commit/8e10548f7f4814e530857d2049d6af6bc78add53
  - 851-006 https://github.com/torvalds/linux/commit/9a4556dad7bd0a6b8339cb72e169f5c76f2af6f1

Cheers

robimarko commented:
When it comes to IPQESS, I will sort that out once 6.1 is in by backporting by then hopefully merged upstream driver.
As far as backports go, ideally 6.1 based ones would be great, I am already using nbd-s pre RC1 6.1 backports.

Ansuel commented:
@john-tho thanks for the massive work...

https://github.com/openwrt/openwrt/pull/11054

if you want to take a look :D 

Also i'm adding the missing patch that you notice...

john-tho commented on package/kernel/linux/modules/hwmon.mk:
```diff
@@ -45,6 +45,7 @@ $(eval $(call KernelPackage,hwmon-ad7418))
 
 define KernelPackage/hwmon-adt7410
   TITLE:=ADT7410 monitoring support
+  DEPENDS:=+LINUX_6_1:kmod-regmap-core
   KCONFIG:= \
 	CONFIG_SENSORS_ADT7X10 \
 	CONFIG_SENSORS_ADT7410
```
This did not function for me unless it was added to the `$(call AddDepends/hwmon` below

john-tho commented on package/kernel/linux/modules/lib.mk:
```diff
@@ -134,6 +134,7 @@ define KernelPackage/lib-zstd
   FILES:= \
 	$(LINUX_DIR)/crypto/zstd.ko \
 	$(LINUX_DIR)/lib/xxhash.ko \
+	$(LINUX_DIR)/lib/zstd/zstd_common.ko.ko@ge6.1 \
```
typo `.ko.ko`?

john-tho commented on target/linux/generic/pending-6.1/486-01-mtd-spinand-add-support-for-ESMT-F50x1G41LB.patch:
```diff
@@ -0,0 +1,143 @@
+ };
+ 
+ /* SPI NAND manufacturers */
++extern const struct spinand_manufacturer esmt_c8_spinand_manufacturer;
+ extern const struct spinand_manufacturer ato_spinand_manufacturer;
```
I think the rest of these are in alphabet order, and noticed it looking at making the 487 patch apply

john-tho commented on target/linux/generic/pending-6.1/486-01-mtd-spinand-add-support-for-ESMT-F50x1G41LB.patch:
```diff
@@ -0,0 +1,143 @@
+@@ -1,3 +1,3 @@
+ # SPDX-License-Identifier: GPL-2.0
+-spinand-objs := core.o ato.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
++spinand-objs := core.o esmt.o ato.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
```
maybe also alphabet order after core here?

dobo90 commented:
> mac80211 backports kernel changes: What is the best approach here? Build a new backports package from 6.1rc1?

FYI, I saw that Felix has done mac80211 and wifi drivers upgrade to wireless-next snapshot (probably it's going to be merged to some rc 6.1 soon). It's on his staging tree:

https://git.openwrt.org/?p=openwrt/staging/nbd.git;a=commit;h=5956944cabe076baf78d9a1a15f38c1d6e4239dc
https://git.openwrt.org/?p=openwrt/staging/nbd.git;a=commit;h=5c374867602896ca6881d662cb9cadc97a3067e7

john-tho commented:
Currently a patch issue with the pending 486 Makefile extra space & 487 Makefile missing ato
```diff
diff --git a/target/linux/generic/pending-6.1/486-01-mtd-spinand-add-support-for-ESMT-F50x1G41LB.patch b/target/linux/generic/pending-6.1/486-01-mtd-spinand-add-support-for-ESMT-F50x1G41LB.patch
index bc769e0dc4..8191c3c61d 100644
--- a/target/linux/generic/pending-6.1/486-01-mtd-spinand-add-support-for-ESMT-F50x1G41LB.patch
+++ b/target/linux/generic/pending-6.1/486-01-mtd-spinand-add-support-for-ESMT-F50x1G41LB.patch
@@ -27,7 +27,7 @@ Signed-off-by: Chuanhong Guo <gch981213@gmail.com>
 @@ -1,3 +1,3 @@
  # SPDX-License-Identifier: GPL-2.0
 -spinand-objs := core.o ato.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
-+spinand-objs := core.o ato.o esmt.o  gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
++spinand-objs := core.o ato.o esmt.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
  obj-$(CONFIG_MTD_SPI_NAND) += spinand.o
 --- a/drivers/mtd/nand/spi/core.c
 +++ b/drivers/mtd/nand/spi/core.c
diff --git a/target/linux/generic/pending-6.1/487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch b/target/linux/generic/pending-6.1/487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch
index 1b5f1faf86..bc7889f560 100644
--- a/target/linux/generic/pending-6.1/487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch
+++ b/target/linux/generic/pending-6.1/487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch
@@ -42,8 +42,8 @@ Submitted-by: Daniel Danzberger <daniel@dd-wrt.com>
 +++ b/drivers/mtd/nand/spi/Makefile
 @@ -1,3 +1,3 @@
  # SPDX-License-Identifier: GPL-2.0
--spinand-objs := core.o esmt.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
-+spinand-objs := core.o esmt.o etron.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
+-spinand-objs := core.o ato.o esmt.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
++spinand-objs := core.o ato.o esmt.o etron.o gigadevice.o macronix.o micron.o paragon.o toshiba.o winbond.o xtx.o
  obj-$(CONFIG_MTD_SPI_NAND) += spinand.o
 --- a/drivers/mtd/nand/spi/core.c
 +++ b/drivers/mtd/nand/spi/core.c
```
---

There are some new tools, generic hack patch & configs commits in master:
https://github.com/Ansuel/openwrt/compare/support-kernel-6.1...openwrt:openwrt:master

---

Apart from ksmbd, have dirty fixes for all the KMODS for aarch64:

- core packages
  - ksmbd https://github.com/openwrt/openwrt/pull/10296 (have not tested)
  - ubnt-ledbar https://github.com/john-tho/openwrt/commit/3de65aa7c7c814f4f22b86f6a63ef92a98e2df53
  - rtl8812au-ct https://github.com/greearb/rtl8812AU_8821AU_linux/pull/4
  - mac80211 backports
    with nbd's 6.1-rc0+2022-10-12, only need to patch 6.1 prandom changes: https://github.com/john-tho/openwrt/commit/96ff3b03cb406a59a1751262335b5019eff6878f from: https://lore.kernel.org/lkml/20221010230613.1076905-1-Jason@zx2c4.com/ linux tree: `git format-patch -n6 197173db990cad244221ba73c43b1df6170ae278`, but these should go away with a newer backports package
- packages
  - ovpn-dco https://github.com/openwrt/packages/pull/19684
  - siit https://github.com/openwrt/packages/pull/19685
  - xr_usb_serial_common https://github.com/openwrt/packages/pull/19686 https://github.com/kasbert/epsolar-tracer/pull/48
  - [xtables-addons](https://inai.de/projects/xtables-addons/) https://github.com/openwrt/packages/pull/19688 https://lore.kernel.org/netfilter-devel/20221023032239.808311-1-git@johnthomson.fastmail.com.au/T/#u
- packages (already mentioned)
  - coova-chilli https://github.com/openwrt/packages/pull/19654
  - libpfring https://github.com/openwrt/packages/pull/19653
- telephony
  - dahdi-linux https://github.com/openwrt/telephony/pull/786 https://github.com/asterisk/dahdi-linux/pull/14

Cheers

abajk commented on target/linux/generic/config-6.1:
```diff
@@ -0,0 +1,7743 @@
+# CONFIG_CPU_IDLE_GOV_TEO is not set
+# CONFIG_CPU_IDLE_MULTIPLE_DRIVERS is not set
+# CONFIG_CPU_ISOLATION is not set
+CONFIG_CPU_LITTLE_ENDIAN=y
```
During the next PR refresh, you can incorporate changes from https://github.com/openwrt/openwrt/commit/9226f1e41948e841c1a546a6b58273e6e994539a.

Ansuel replied:
Thanks for the remainder. I will do this!

Ansuel replied:
@abajk should we disable the option in generic config and put them in each required target?

1715173329 commented on package/kernel/linux/modules/netfilter.mk:
```diff
@@ -1148,21 +1148,44 @@ endef
   TITLE:=Netfilter nf_tables routing/NAT offload support
-  DEPENDS:=@IPV6 +kmod-nf-flow +kmod-nft-nat
+  DEPENDS:=@IPV6 +kmod-nf-flow +kmod-nft-nat \
+	+kmod-nft-offload-ipv4 +kmod-nft-offload-ipv6
```
this will results firewall4 unavailable on kernel 6.1 (`firewall4` depends on `nft-offload`, and `nft-offload` depends on `nft-offload-ipv4/v6` which requires `!LINUX_6_1`).

robimarko replied:
Nice catch, why are there even being added here as new dependencies as their respective kmods got squashed into one upstream?

Ansuel replied:
How should this be solved?

john-tho replied:
I made this change (against master) to nft-offload, and it seems to work for me?

```diff
---
 package/kernel/linux/modules/netfilter.mk | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/package/kernel/linux/modules/netfilter.mk b/package/kernel/linux/modules/netfilter.mk
index e6e2c9b637..7e17458ed7 100644
--- a/package/kernel/linux/modules/netfilter.mk
+++ b/package/kernel/linux/modules/netfilter.mk
@@ -1154,15 +1154,15 @@ define KernelPackage/nft-offload
   DEPENDS:=@IPV6 +kmod-nf-flow +kmod-nft-nat
   KCONFIG:= \
 	CONFIG_NF_FLOW_TABLE_INET \
-	CONFIG_NF_FLOW_TABLE_IPV4 \
-	CONFIG_NF_FLOW_TABLE_IPV6 \
+	CONFIG_NF_FLOW_TABLE_IPV4@lt5.17 \
+	CONFIG_NF_FLOW_TABLE_IPV6@lt5.17 \
 	CONFIG_NFT_FLOW_OFFLOAD
   FILES:= \
 	$(LINUX_DIR)/net/netfilter/nf_flow_table_inet.ko \
-	$(LINUX_DIR)/net/ipv4/netfilter/nf_flow_table_ipv4.ko \
-	$(LINUX_DIR)/net/ipv6/netfilter/nf_flow_table_ipv6.ko \
+	$(LINUX_DIR)/net/ipv4/netfilter/nf_flow_table_ipv4.ko@lt5.17 \
+	$(LINUX_DIR)/net/ipv6/netfilter/nf_flow_table_ipv6.ko@lt5.17 \
 	$(LINUX_DIR)/net/netfilter/nft_flow_offload.ko
-  AUTOLOAD:=$(call AutoProbe,nf_flow_table_inet nf_flow_table_ipv4 nf_flow_table_ipv6 nft_flow_offload)
+  AUTOLOAD:=$(call AutoProbe,nf_flow_table_inet nf_flow_table_ipv4@lt5.17 nf_flow_table_ipv6@lt5.17 nft_flow_offload)
 endef
 
 $(eval $(call KernelPackage,nft-offload))
-- 
2.38.1


```

robimarko commented:
I would suggest using `@!LINUX_5_10` for anything that does not work on 5.10 only as otherwise new kernel bump will require adding `LINUX_6_1` to the supported list of each kmod again.

robimarko committed:
generic: 6.1: add missing symbols

These symbols got exposed once building for ipq807x.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
kernel: modules: fs: adapt for kernel 6.1

Adapt filesystem kmods for building under kernel 6.1:
* Depend on kernel not being 5.10 rather than only 5.15
* kmod-fs-9p depends on kmod-fs-netfs from 5.17 as they started using
netfs helpers
* Set new KConfig options to N

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
@Ansuel Finally got around to 6.1 again, so I have some more kmod and generic config fixups for 6.1.

https://github.com/robimarko/openwrt/commit/d7a8044053b085065b217c2a1a8a27af19f44df0
https://github.com/robimarko/openwrt/commit/d21231e45dbac2d1ef75aacd6b409227a449f08d
and
https://github.com/robimarko/openwrt/commit/3f4bb3d3d86b68e6fbd8e6e60e46e167622ac359

Ansuel commented:
@robimarko did you manage to make 6.1 work with ipq807x? just curious

robimarko commented:
Yeah, look at the same branch, only ath11k is broken but that is a QMI issue that I have seen in Buildroot as well for a while and bisected to the exact broken commit but I have no clue why that commit broke it.

john-tho commented on target/linux/generic/pending-6.1/497-mtd-mtdconcat-add-dt-driver-for-concat-devices.patch:
```diff
@@ -165,7 +165,7 @@ Signed-off-by: Bernhard Frauendienst <kernel@nospam.obeliks.de>
 +
 +	of_for_each_phandle(&it, err, node, "devices", NULL, 0) {
-+		mtd = of_get_mtd_device_by_node(it.node);
++		mtd = get_mtd_device_by_node(it.node);
```
bad change here?
```
  CC      drivers/mtd/composite/virt_concat.o
drivers/mtd/composite/virt_concat.c: In function 'virt_concat_probe':
drivers/mtd/composite/virt_concat.c:80:23: error: implicit declaration of function 'get_mtd_device_by_node'; did you mean 'of_get_mtd_device_by_node'? [-Werror=implicit-function-declaration]
   80 |                 mtd = get_mtd_device_by_node(it.node);
      |                       ^~~~~~~~~~~~~~~~~~~~~~
      |                       of_get_mtd_device_by_node
drivers/mtd/composite/virt_concat.c:80:21: warning: assignment to 'struct mtd_info *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
   80 |                 mtd = get_mtd_device_by_node(it.node);
      |                     ^
cc1: some warnings being treated as errors
```

PolynomialDivision commented:
Can you also bump mac80211? ;)

PolynomialDivision commented:
I bumped the mediatek version, however the bluetooth kernel modules fails currently with:
```
ERROR: module '/home/nick/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/linux-6.1-rc2/drivers/bluetooth/btusb.ko' is missing.
```

https://github.com/PolynomialDivision/openwrt/commits/bump-mt7622

PolynomialDivision commented:
How do you test everything without being able to compile mac80211?

Ansuel commented:
@PolynomialDivision the wifi part in a router is optional currently this is only boot tested with dsa driver

PolynomialDivision commented:
Ah okay thanks! @chunkeey wrote a wile ago that you can also compile without the backports directly in kernel drivers:
https://forum.openwrt.org/t/do-i-really-need-mac80211-backports-package/15585/5
However the link does not work. :/

Ansuel commented:
@PolynomialDivision we tend to prefer backport as we can have better handling with patches and stuff I assume to use mac80211 we have to apply all the mac80211 patches to the generic target + a big .mk in the modules dir

PolynomialDivision commented:
@Ansuel It seems like someone did that already. I want to test on a recent mac80211 and kernel for up-streaming certain wireless driver additions. :) What about creating a new backport for 6.1rc? (cc @hauke )

PolynomialDivision commented:
I also cleaned the mac80211:
- https://github.com/openwrt/openwrt/pull/11179
- https://github.com/openwrt/openwrt/pull/11180
- https://github.com/openwrt/openwrt/pull/11181
- https://github.com/openwrt/openwrt/pull/11182
- https://github.com/openwrt/openwrt/pull/11183

john-tho commented:
An 11 patch series went into net-next which may replace pending/724 & friends:
https://github.com/openwrt/openwrt/blob/master/target/linux/generic/pending-5.15/724-net-mtk_sgmii-implement-mtk_pcs_ops.patch
https://lore.kernel.org/netdev/Y1qDMw+DJLAJHT40@shell.armlinux.org.uk/T/#u
first & last: https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit?id=9c5a170677c3c8facc83e931a57f4c99c0511ae0 https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit?id=81b0f12a2a8a1699a7d49c3995e5f71e4ec018e6
I don't have an mtk_eth_soc device with SGMII to test. That series was [tested & built upon a bpi-r3](https://lore.kernel.org/netdev/20221020144431.126124-1-linux@fw-web.de/T/#u)

Mostly offtopic: I am having some trouble with my ramips mt7621 mtk_eth_soc gmac1 RGMII to AR8033, turns out since OpenWrt's 5.15 kernel. It (only) transmits. Looks like `ethtool -S sfp` `rx_fcs_errors:` are increasing when I am expecting to receive a packet.
Along the way found some upstream 6.1rc kernel issues: [mips boot/compressed compile](https://lore.kernel.org/linux-mips/20221031170749.2159430-1-git@johnthomson.fastmail.com.au/), [mt7621 PCI oops](https://lore.kernel.org/lkml/20221104205242.3440388-1-git@johnthomson.fastmail.com.au/T/#u), and [mt7621 boot kzalloc](https://lore.kernel.org/lkml/20221103050538.1930758-1-git@johnthomson.fastmail.com.au/T/#u)

@PolynomialDivision If you do want to test mac80211, nbd's staging git has a backports 6.1 pre-rc1, and it compiles after the later prandom changes. Details are in the comments above.

The master tools & dnsmasq changes would be nice to have when you find some spare time to rebase. Big thanks.

PolynomialDivision commented:
> @PolynomialDivision If you do want to test mac80211, nbd's staging git has a backports 6.1 pre-rc1, and it compiles after the later prandom changes. Details are in the comments above.

@john-tho Thanks! :) What do you mean with prandom changes? Exactly here it fails for me with:
```
Building backport-include/backport/autoconf.h ... done.
  CC [M]  /home/nick/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/backports-6.1-rc0+2022-10-12/net/mac80211/rc80211_minstrel_ht.o
/home/nick/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/backports-6.1-rc0+2022-10-12/net/mac80211/rc80211_minstrel_ht.c: In function 'init_sample_table':
/home/nick/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/backports-6.1-rc0+2022-10-12/net/mac80211/rc80211_minstrel_ht.c:1995:17: error: implicit declaration of function 'prandom_bytes'; did you mean 'get_random_bytes'? [-Werror=implicit-function-declaration]
 1995 |                 prandom_bytes(rnd, sizeof(rnd));
      |                 ^~~~~~~~~~~~~
      |                 get_random_bytes
cc1: some warnings being treated as errors
```

mj22226 commented:
@PolynomialDivision  I backported some patches to compile the current backports  version on 6.1 https://github.com/mj22226/openwrt/commit/7aac779b28157a0675b19c3b3f63db072a0243b6

PolynomialDivision commented:
@Ansuel Can you rebase on master? You are missing some kernel modules.

Ansuel commented:
Hi, sorry for the delay busy with life and debugging ipq806x sooo i'm giving this low priority :(

Ansuel commented:
@john-tho can you help me with this? :D 

@PolynomialDivision :D finally found time to refresh this.

john-tho commented:
Nice! A minimal image compiles & boots for me on my mt7621 router (with my target specific changes).

Some outstanding previous comments:
- [netfilter.mk fix nft-offload](https://github.com/openwrt/openwrt/pull/11011#pullrequestreview-1155943404): adding `@lt5.17` after the v4 & v6 in KCONFIG, FILES, & AUTOLOAD from nft-offload worked for me, and removed the need for the extra packages
- [robimarko's three changes](https://github.com/openwrt/openwrt/pull/11011#issuecomment-1291955639): especially the fs.mk fs-netfs fixes from https://github.com/robimarko/openwrt/commit/3f4bb3d3d86b68e6fbd8e6e60e46e167622ac359
- [invert @LINUX_5_15 to not 5_10](https://github.com/openwrt/openwrt/pull/11011#issuecomment-1291821376): need to check & test the TPM package in other.mk
- [pending/497-mtd-mtdconcat-add-dt-driver-for-concat-devices.patch missing `of_` from `get_mtd_device_by_node`](https://github.com/openwrt/openwrt/pull/11011#pullrequestreview-1157653903)
- [pending/724 & friends looks to have been taken with some changes upstream](https://github.com/openwrt/openwrt/pull/11011#issuecomment-1306791781)

Ansuel commented:
@john-tho we should really summarize and catch up if we missed something down the line...

john-tho commented:
Found that my wan firewall rules were not loading:
```
root@OpenWrt:/usr/share/ucode# fw4 -v check
Section @zone[1] (wan) option 'mtu_fix' requires kernel 4.10 or later
Section @zone[1] (wan) tested: this.kernel '0' < 0x040a0000
Section @rule[0] (Allow-DHCP-Renew) option 'src' specifies invalid value 'wan'
```

`fw4` reads /proc/version: https://git.openwrt.org/?p=project/firewall4.git;a=blob;f=root/usr/share/ucode/fw4.uc;h=5dce90dd09c2657f8fcc6dda0dfddb4f4ad13f91;hb=HEAD#l438

Our kernel sublevel is 0, but in an OpenWrt-ism this is not shown in `KERNELVERSION`. kernel normally includes it: `cat include/config/kernel.release: 6.1.0-rc8`
```
cat /proc/version 
Linux version 6.1 (john@john) (mipsel-openwrt-linux-musl-gcc (OpenWrt GCC 11.3.0 r19724+16-1521d5f453) 11.3.0, GNU ld (GNU Binutils) 2.37) #0 SMP Wed Dec 14 02:59:13 2022
```

I first adjusted fw4.uc, but then decided that I would prefer to match what kernel does, so went headbutting Makefiles, and came up with two commits:
- https://github.com/john-tho/openwrt/commit/1ebba2fe6f6d26631ae01fe3cfe74057606b58cb allow specifying Openwrt kernel version as `= .0-rc2`, or = `.0`, and fix the source and and directory name for linux tarballs.
- https://github.com/john-tho/openwrt/commit/d3e65c0420e16a93542320b472e35b1391110b3a expand `=-rc2` to `.0-rc2` and `=` to `.0`

Likely there are much better approaches than this, but it _seems_ to work for me. Not especially applicable to OpenWrt master, as it will never include a `-rc` or `.0` sublevel release, but it helps with my poking around.

Cheers

Ansuel commented:
@john-tho i mean it's an interesting bug

Neustradamus commented:
To follow :)

Note, currently it is 6.1.1 (2022-12-21)
- https://www.kernel.org

castillofrancodamian commented:
I don't see that 6.1 is LTS.

Ansuel commented:
It will as the year will end... There is 0 possibility 6.2 will go out of
RC stage in 3 days ahahahah

Il Gio 29 Dic 2022, 17:17 Franco Castillo ***@***.***> ha
scritto:

> I don't see that 6.1 is LTS.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/11011#issuecomment-1367441326>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQR4RQ7WVRZZSUVDA2TWPW2S5ANCNFSM6AAAAAARIQXNCM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

castillofrancodamian commented:
Yes or does there have to be LTS in each year?

Ansuel commented:
afaik yes

robimarko commented:
Yeah, last kernel release of the year is LTS, that's been the practice for a long time

t-8ch commented:
There has been some discussion on LKML: https://lore.kernel.org/lkml/Y53BputYK+3djDME@kroah.com/
Maybe this PR already generated some insights that could be reported to Greg?

robimarko commented:
Personally, no idea which insights that are helpful can we share.
What Greg is pretty much asking is for one of the major companies using LTS releases to contribute back by at least testing, ideally by having a person to help with the maintenance and not just freeloading.

And all he got was the usual I want XYZ release to be LTS because there isn't XYZ which is a major breaking change for me though it doesn't affect me at all, for example, rust support which doesn't influence anybody pretty much.

I doubt that LF will break the tradition of picking the last year release as the LTS, picking 6.0 would be rather crazy now since support for it is pretty much over.

Ansuel commented:
In addition to that the complain about rust is stupid considering how long
it took and the fact that even Linus helped and wanted that. But we will
see how it goes... For sure migrating from 6.1 to 6.0 will be a PITA for me.

Il Gio 29 Dic 2022, 21:46 Robert Marko ***@***.***> ha
scritto:

> Personally, no idea which insights that are helpful can we share.
> What Greg is pretty much asking is for one of the major companies using
> LTS releases to contribute back by at least testing, ideally by having a
> person to help with the maintenance and not just freeloading.
>
> And all he got was the usual I want XYZ release to be LTS because there
> isn't XYZ which is a major breaking change for me though it doesn't affect
> me at all, for example, rust support which doesn't influence anybody pretty
> much.
>
> I doubt that LF will break the tradition of picking the last year release
> as the LTS, picking 6.0 would be rather crazy now since support for it is
> pretty much over.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/11011#issuecomment-1367577157>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQRZ4FCOP6EUWLMUDE3WPX2DDANCNFSM6AAAAAARIQXNCM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

robimarko commented:
We will see, they will decide soon, 6.0 is not ideal for anybody as everybody is counting on 6.1 being LTS

Ansuel commented:
Honestly everyone wants to jump on the rust train and having an lts kernel
with that is actually beneficial to everyone

Il Gio 29 Dic 2022, 21:58 Robert Marko ***@***.***> ha
scritto:

> We will see, they will decide soon, 6.0 is not ideal for anybody as
> everybody is counting on 6.1 being LTS
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/11011#issuecomment-1367581853>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQRQN5RDBPFYFDEH2WDWPX3QDANCNFSM6AAAAAARIQXNCM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

robimarko commented:
Rust support in the kernel is so minor that it's irrelevant as nothing in the kernel depends on it, it's just the first step.

john-tho commented:
When you get a chance, a rebase would be appreciated for the chunky toolchain updates for GCC & binutils. A few extra backports as well.
Thank you!

Neustradamus commented:
About Linux Kernel 6.1 and LTS, a lot of messages:
- https://lore.kernel.org/lkml/Y8HPw2t+TbdXa83C@mit.edu/

Ansuel commented:
Finally https://www.phoronix.com/news/Linux-6.1-LTS-Official

Spudz76 commented:
rebase goes fairly well until the end but I think the .github change can just be removed since the whole JSON thing was redone

now I'm trying to figure out how to get ramips/mt7621 to build with LINUX_6_1 set, I swear I knew how before LINUX_5_15 was enabled for it

rmandrad commented:
any chance of getting ipq807x on the list of targets for this PR? or does anyone feel that due to the 5.10 dependency we will jump over 6.1 to the next LTS ?

Ansuel commented:
@rmandrad i will update this when we branch for 23.xx 

This is really just for reference for other maintainers if they want to work on other target for 6.1 support 

I feel like it's wasted time updating this from time to time since we are close to branching and many changes are happening

rmandrad commented:
makes sense thx for the reply

tmn505 commented:
@Ansuel I rebased part of this PR in my tree (https://github.com/tmn505/openwrt/commits/zx10-k6.1). Where the changes were more intrusive, I added my SoB. Feel free to pick it up.

mrkiko commented:
Great work @Ansuel ! :)

mrkiko commented:
Oops, wrong PR - but great work no matter :)

Neustradamus commented:
Officially LTS since 6.1.12 Kernel :)
- https://kernel.org/

timocapa commented:
> now I'm trying to figure out how to get ramips/mt7621 to build with LINUX_6_1 set, I swear I knew how before LINUX_5_15 was enabled for it

let me know should you get there, i can test with 2 mt7621 devices (assuming they boot)

Bartel-C8 commented:
> now I'm trying to figure out how to get ramips/mt7621 to build with LINUX_6_1 set, I swear I knew how before LINUX_5_15 was enabled for it

This could help? 
https://github.com/openwrt/openwrt/commit/77d061167cf8a7edb3ea7cbc9926f25c8ac8fe1e

CHKDSK88 committed:
generic: kernel: add missing symbol

Add missing symbol while adding layerscape support

Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>
[ add commit description ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

guidosarducci commented:
@Ansuel Thanks for this work so far!

I've added support for malta on top of your branch. Feel free to pull in the top 3 commits from https://github.com/guidosarducci/openwrt/commits/master-support-malta-6.1 when the time comes...

Ansuel committed:
kernek: use struct_group to wipe ar8216 volatile priv data

Instead of reference vlan and do strange subtraction, use the handy
struct_group() to create a virtual struct of the same size of the
members. This permits to have a more secure memset and fix compilation
warning in 6.1 where additional checks are done.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: manually refresh mglru patch with new kernel version

Refresh mglru patch for new kernel version due to very fun backport
changing name from folio to pages.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: drop pending patch backported as fixes

Drop pending patch backported as fixes in later kernel version.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@john-tho finally branched and finally updated to 6.1.29 :D 
@robimarko if you want to have some fun

guidosarducci commented:
@Ansuel Nice to see this all getting closer!  For those interested, I've now included both `malta` and `armvirt` support in my [master-support-malta-6.1](https://github.com/guidosarducci/openwrt/commits/master-support-malta-6.1) branch based on your last version.

Ansuel committed:
kernel: use eth_random_addr instead of random_ether_addr

random_ether_addr is just a reference to eth_random_addr, that was later
dropped in more recent kernel version.

Drop random_ether_addr and use eth_random_addr directly to fix
compilation error in 6.1

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: refresh backport patches

Refresh backport patches with make target/linux/refresh.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: refresh hack patches

Refresh hack patches with make target/linux/refresh.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: 6.1: refresh pending patches

Refresh pending patches with make target/linux/refresh.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: 6.1: refresh kernel patches

Refresh ipq806x kernel patches with make target/linux/refresh.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

robimarko commented:
What is left to do?

Can I help with anything?

Ansuel commented:
@robimarko just merged... i expect more fixup in kernel modules and other small fix later when other target and 6.1 will get more attention in general.

robimarko commented:
Yeah, I just posted the comment and then got the notification, nice work.

As always things will float once its starts being used, will prepare ipq807x.
